### ISSUE TITLE
Clippy stops working after printing its version unless I modify the source code

### ISSUE THREAD

**Issue by Guilherme-Vasconcelos** [rust-lang/rust-clippy#9190]
### Summary

I realized that if I run `cargo clippy` with `-- -Vv` flag, clippy completely stops working until I either change my source code, or `cargo clean`.

### Reproducer

I tried this code:

```rust
/// main.rs

fn main() {
    let t = 10;
}
```

Then I ran `cargo clippy` and got a warning:

```
warning: unused variable: `t`
 --> src/main.rs:2:6
  |
2 |     let t = 10;
  |         ^ help: if this is intentional, prefix it with an underscore: `_t`
  |
  = note: `#[warn(unused_variables)]` on by default
```

Then I ran `cargo clippy -- -Vv` and got my clippy version:

```
rustc 1.62.0 (a8314ef7d 2022-06-27)
binary: rustc
commit-hash: a8314ef7d0ec7b75c336af2c9857bfaf43002bfc
commit-date: 2022-06-27
host: x86_64-unknown-linux-gnu
release: 1.62.0
LLVM version: 14.0.5
    Finished dev [unoptimized + debuginfo] target(s) in 0.01s
```

And when I ran `cargo clippy` again, it no longer emits that warning:

```
$ cargo clippy
    Finished dev [unoptimized + debuginfo] target(s) in 0.00s
```

I expected the last `cargo clippy` to have the same warning as before (unused variable).

If I modify `src/main.rs` or do a `cargo clean`, clippy starts working normal again.

### Version

```text
rustc 1.62.0 (a8314ef7d 2022-06-27)
binary: rustc
commit-hash: a8314ef7d0ec7b75c336af2c9857bfaf43002bfc
commit-date: 2022-06-27
host: x86_64-unknown-linux-gnu
release: 1.62.0
LLVM version: 14.0.5
```
